AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/org/androidannotations/rootfragment/RootFragmentTest.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) 2016-2019 the AndroidAnnotations project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed To in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.androidannotations.rootfragment;
+
+import org.androidannotations.internal.AndroidAnnotationProcessor;
+import org.androidannotations.testutils.AAProcessorTestHelper;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RootFragmentTest extends AAProcessorTestHelper {
+
+	@Before
+	public void setUp() {
+		addManifestProcessorParameter(RootFragmentTest.class);
+		addProcessor(AndroidAnnotationProcessor.class);
+	}
+
+	@Test
+	public void eBeanOnFragmentCompiles() {
+		assertCompilationSuccessful(compileFiles(SomeBeanWithRootFragment.class));
+	}
+
+	@Test
+	public void eBeanOnCustomFragmentCompiles() {
+		assertCompilationSuccessful(compileFiles(SomeBeanWithRootFragmentWithCustomFragment.class));
+	}
+
+	@Test
+	public void eBeanOnCustomSupportFragmentsCompiles() {
+		assertCompilationSuccessful(compileFiles(defPath("support/Fragment.java", "CustomSupportFragment.java", "SomeBeanWithRootFragmentWithCustomSupportFragment.java")));
```

Comment:
Why is this needed? I see similar tests, for example `ActivityIntentFragmentTest`, which is using `toPath()`.

Reply:
```java
CompileResult result = compileFiles(toPath(ActivityIntentFragmentTest.class, "support/Fragment.java"), //
				toPath(ActivityIntentFragmentTest.class, "support/ActivityCompat.java"), ActivityInManifest.class);
```

Why did you add the `defPath()` method?

Reply:
@WonderCsabo , "defPath" accepts multiple paths (which is needed when having several "support-dependent files".. Also, it simplifies having to pass "the current class" per each call to "toPath", it is a little improvement, without it, when you have something like three files (like in that example above), basically you have to make a huge line with "toPath" and passing the test class to that method  (so 3 times). That doesn't seem that clean. 

"def" just assumes that the "package-holder class" is the same than the one of the test

Reply:
I see it now, thanks for improving it!